Sources/Nimble/Utils/Async.swift

```diff
@@ -351,4 +358,12 @@ internal func pollBlock(
         }.timeout(timeoutInterval, forcefullyAbortTimeout: timeoutInterval / 2.0).wait(fnName, file: file, line: line)
 
         return result
+    #else
+        do {
+            let pass = try expression()
+            return pass ? .Completed(true) : .TimedOut
+        } catch let error {
+            return .ErrorThrown(error)
+        }
```

Comment:
Excellent point. When I was first hacking on this I was looking for the smallest change set that would satisfy the compiler, but you're right that non-functional features should be entirely unavailable.